### PR TITLE
Support specific config source async updating

### PR DESCRIPTION
**Pull request by theopolis**

### CODE REVIEW

**Comment by theopolis** [comment]
ok to test

**Comment by theopolis** [comment]
This last update changes "scheduledQueries" to a map using the name as a key. After a bit of digging, it seems that the name is used as a key in RocksDB, so duplicate keys mean undefined behavior when considering multiple config sources (in a single config Plugin). 

The new syntax for adding a schedule is:

```json
{
  "schedule": {
    "kernel_extensions": {"query": "select * from kernel_extensions", "interval": 60}
  }
}
```

The code supports the proposed legacy format of `"scheduledQueries": []`, but it is not recommended anymore.

**Comment by theopolis** [comment]
ok to test

**Comment by theopolis** [comment]
ok to test

**Comment by wxsBSD** [comment]
Apologies if I missed it but would it be nice to start issuing a warning when "scheduledQueries" is used? Give some advance notice that it will be removed in the future?

**Comment by theopolis** [comment]
We'll never remove it :) but we'll update the documentation to only include the map-based configuration. That's a great suggestion to issue an INFO log line too.

File: osquery/config/config.cpp [hunk 1]
```diff
@@ -95,76 +91,80 @@ inline void mergeAdditional(const tree_node& node, OsqueryConfig& conf) {
...
-    }
+  pt::ptree tree;
+  pt::read_json(json_data, tree);
 
-    if (tree.count("additional_monitoring") > 0) {
-      for (const auto& node : tree.get_child("additional_monitoring")) {
-        mergeAdditional(node, conf);
-      }
+  // Legacy query schedule vector support.
+  if (tree.count("scheduledQueries") > 0) {
```
Comment (theopolis) [inline osquery/config/config.cpp]: Output LOG(INFO) line about deprecated key.

File: include/osquery/database/results.h [hunk 2]
```diff
@@ -271,4 +270,123 @@ bool addUniqueRowToQueryData(QueryData& q, const Row& r);
...
+
+  /// The time that the query was executed, in unix time
+  int unixTime;
+
+  /// The time that the query was executed, in ASCII
+  std::string calendarTime;
+
+  /// equals operator
+  bool operator==(const ScheduledQueryLogItem& comp) const {
+    return (comp.diffResults == diffResults) && (comp.name == name);
```
Comment (marpaia) [inline include/osquery/database/results.h]: Do we not care about the other struct fields?

**Comment by marpaia** [comment]
ship it!

**Comment by theopolis** [comment]
ok to test

**Pull request merged**